File: backend/models/Message.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

// this is the Message schema
const Message = new Schema({
    id: {
        type: Number,
        required: true
    },
    created: {
        type: Date,
        default: Date.now()
    },
    sender: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    content: {
        type: String,
        nullable: true
    },
});

module.exports = mongoose.model('Message', Message);

File: chatapp/app/src/main/java/com/example/chatapp/Schemes/Message.java
package com.example.chatapp.Schemes;

import java.util.Date;

public class Message {
    private int id;
    private Date created;
    private User sender;
    private String content;


    public Message(int id, Date created, User sender, String content) {
        this.id = id;
        this.created = created;
        this.sender = sender;
        this.content = content;
    }

}
